scenarioo-server/src/main/java/org/scenarioo/dao/search/elasticsearch/ElasticSearchSearcher.java

```diff
@@ -45,93 +47,96 @@
-            this.client = client;
-            this.indexName = indexName;
-
-			useCaseReader = generateStandardReaders(UseCase.class, SearchableUseCase.class);
-			scenarioReader = generateStandardReaders(Scenario.class, SearchableScenario.class);
-			stepReader = generateStandardReaders(StepDescription.class, SearchableStep.class);
-    }
-
-    SearchResults search(final SearchRequest searchRequest) {
-        SearchResponse searchResponse = executeSearch(searchRequest);
-
-        if (searchResponse.getHits().getHits().length == 0) {
-            LOGGER.debug("No results found for " + searchRequest);
-            return SearchResults.noHits();
-        }
-
-        SearchHit[] hits = searchResponse.getHits().getHits();
-
-        List<SearchableObject> results = new ArrayList<>();
-        for (SearchHit searchHit : hits) {
-            try {
-                String type = searchHit.getType();
-				switch (type) {
-					case FullTextSearch.USECASE:
-						results.add(parseUseCase(searchHit));
-						break;
-					case FullTextSearch.SCENARIO:
-						results.add(parseScenario(searchHit));
-						break;
-					case FullTextSearch.STEP:
-						results.add(parseStep(searchHit));
-						break;
-					default:
-						LOGGER.error("No type mapping for " + searchHit.getType() + " known.");
-						break;
+	private final String indexName;
+	private final RestHighLevelClient restClient;
+
+	private final ObjectReader useCaseReader;
+	private final ObjectReader scenarioReader;
+	private final ObjectReader stepReader;
+
+	ElasticSearchSearcher(final String indexName, RestHighLevelClient restClient) {
+		this.indexName = indexName;
+		this.restClient = restClient;
+
+		useCaseReader = generateStandardReaders(UseCase.class, SearchableUseCase.class);
+		scenarioReader = generateStandardReaders(Scenario.class, SearchableScenario.class);
+		stepReader = generateStandardReaders(StepDescription.class, SearchableStep.class);
+	}
+
+	SearchResults search(final SearchRequest searchRequest) {
+		SearchResponse searchResponse;
+		try {
+			searchResponse = executeSearch(searchRequest);
+		} catch (Exception e) {
+			LOGGER.error("Search failed", e);
+			return SearchResults.noHits();
+		}
+
+		if (searchResponse.getHits().getHits().length == 0) {
+			LOGGER.debug("No results found for " + searchRequest);
+			return SearchResults.noHits();
+		}
+
+		SearchHit[] hits = searchResponse.getHits().getHits();
+
+		List<SearchableObject> results = new ArrayList<>();
+		for (SearchHit searchHit : hits) {
+			try {
+				String stringRepresentation = searchHit.toString();
+				if (stringRepresentation.contains("\"type\" : \"" + FullTextSearch.SCENARIO + "\"")) {
+					results.add(parseScenario(searchHit));
+				} else if (stringRepresentation.contains("\"type\" : \"" + FullTextSearch.USECASE + "\"")) {
+					results.add(parseUseCase(searchHit));
+				} else if (stringRepresentation.contains("\"type\" : \"" + FullTextSearch.STEP + "\"")) {
```

Comment:
would be better if we could do proper parsing of the type here - could be error prone for different setups or version changes

Reply:
Introduced proper field "type" in SearchableObject which allows us to do a safe switch case to discover the correct type of the search result.